connectivity-exporter/metrics/metrics.go

```diff
@@ -56,19 +57,80 @@ func Apply(ctx context.Context, wg *sync.WaitGroup, incs <-chan *Inc, snapshots
 }
 
 func applyInc(inc *Inc) {
-	seconds.WithLabelValues("clock", inc.SNI).Add(inc.AllSeconds)
-	seconds.WithLabelValues("active", inc.SNI).Add(inc.ActiveSeconds)
-	seconds.WithLabelValues("failed", inc.SNI).Add(inc.FailedSeconds)
-	seconds.WithLabelValues("active_failed", inc.SNI).Add(inc.ActiveFailedSeconds)
-	connections.WithLabelValues("successful", inc.SNI).Add(inc.SuccessfulConnections)
-	connections.WithLabelValues("unacknowledged", inc.SNI).Add(inc.UnacknowledgedConnections)
-	connections.WithLabelValues("rejected", inc.SNI).Add(inc.RejectedConnections)
-	connections.WithLabelValues("rejected_by_client", inc.SNI).Add(inc.RejectedConnectionsByClient)
-	packets.WithLabelValues("orphan", inc.SNI).Add(inc.OrphanPackets)
+	inc.SNI.refreshTTL <- nil
+	seconds.WithLabelValues("clock", inc.SNI.name).Add(inc.AllSeconds)
+	seconds.WithLabelValues("active", inc.SNI.name).Add(inc.ActiveSeconds)
+	seconds.WithLabelValues("failed", inc.SNI.name).Add(inc.FailedSeconds)
+	seconds.WithLabelValues("active_failed", inc.SNI.name).Add(inc.ActiveFailedSeconds)
+	connections.WithLabelValues("successful", inc.SNI.name).Add(inc.SuccessfulConnections)
+	connections.WithLabelValues("unacknowledged", inc.SNI.name).Add(inc.UnacknowledgedConnections)
+	connections.WithLabelValues("rejected", inc.SNI.name).Add(inc.RejectedConnections)
+	connections.WithLabelValues("rejected_by_client", inc.SNI.name).Add(inc.RejectedConnectionsByClient)
+	packets.WithLabelValues("orphan", inc.SNI.name).Add(inc.OrphanPackets)
 }
 
 func applySnapshot(snapshot promextra.Snapshot) {
 	if err := execution.ApplySnapshot(snapshot); err != nil {
 		klog.Error("failed to apply snapshot", err)
 	}
 }
+
+func NewSNI(sni string) *SNI {
+	s := &SNI{
+		refreshTTL: make(chan interface{}),
+		name:       sni,
+	}
+	return s
+}
+
+type TTLTimer interface {
+	getChannel() <-chan time.Time
+	resetTimer()
+	cleanup()
+}
+
+type RealTimer struct {
+	Timer *time.Timer
+}
+
+func (t *RealTimer) getChannel() <-chan time.Time {
+	return t.Timer.C
+}
+
+func (t *RealTimer) resetTimer() {
+	t.Timer.Stop()
+	t.Timer.Reset(TTL)
+}
+
+func (t *RealTimer) cleanup() {
+	t.Timer.Stop()
+}
+
+func (sni *SNI) StartTTL(t TTLTimer) {
```

Comment:
Since this function is always called as a go routine (e.g. `go snis[sni].StartTTL(...`), maybe it could create the go routine itself instead of letting callers do it?